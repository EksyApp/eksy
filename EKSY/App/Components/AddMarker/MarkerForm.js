import React, {Component} from 'react'
import PointSelector from "../Common/PointSelector";
import {View, StyleSheet} from "react-native";
import TextInputArea from "../Common/TextInputArea";
import ImagePicker from "../Common/ImagePicker";
import CompactPictureList from "../Common/CompactPictureList";
import CheckBoxList from "../Common/CheckBoxList";
import Message from "../Common/Message";
import Input from "../Common/Input";

export default class MarkerForm extends Component {
	
	constructor(props) {
		super(props)
		
		this.state = {
			imageResponse: ""
		}
	}
	
	render() {
		return (
				<View>
						<View style={styles.selectorContainer}>
							<PointSelector
									style={styles.selector}
									initialRegion={this.props.initialRegion}
							    onChange={(region) => {this.props.onRegionChange(region)}}
							/>
						</View>
						<View>
							<Input
									label="Title"
							    defaultValue={this.props.initialTitle}
							    onChangeText={(title) => {this.props.onTitleChange(title)}}
							/>
							<TextInputArea
									label="Text"
							    defaultValue={this.props.initialText}
							    onChangeText={(text) => {this.props.onTextChange(text)}}
							/>
							<Message>{this.state.imageResponse}</Message>
							<ImagePicker
									buttonText="Add image"
									onNewImage={(image) => {
										this.setState({imageResponse: "Image added!"})
										this.props.onNewImage(image)
									}}
									onPickerError={(error) => this.setState({imageResponse: error.message})}
									onUriError={(error) => this.setState({imageResponse: error.message})}
							/>
							<CompactPictureList
									data={this.props.images}
							/>
							<CheckBoxList
									data={this.props.filters}
									onPress={(name, checked) => {this.props.onFilterChange(name, checked)}}
									titleKey="addingDescription"
							/>
						</View>
				</View>
		)
	}
	
}
const styles = StyleSheet.create({
	
	selectorContainer: {
		height: 300,
		width: '100%'
	},
	
	selector: {
		width: '100%',
		height: '100%'
	}
	
})