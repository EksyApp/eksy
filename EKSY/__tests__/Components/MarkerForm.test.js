import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import MarkerForm from "../../App/Components/AddMarker/MarkerForm";

jest.mock('react-native-maps', () => {
	const React = require.requireActual('react');
	const MapView = require.requireActual('react-native-maps');
	
	class MockCallout extends React.Component {
		render() {
			return React.createElement('Callout', this.props, this.props.children);
		}
	}
	
	class MockMarker extends React.Component {
		render() {
			return React.createElement('Marker', this.props, this.props.children);
		}
	}
	
	class MockMapView extends React.Component {
		render() {
			return React.createElement('MapView', this.props, this.props.children);
		}
	}
	
	class MockMapViewAnimated extends React.Component {
		render() {
			return React.createElement('MapView', this.props, this.props.children);
		}
	}
	
	MockCallout.propTypes = MapView.Callout.propTypes;
	MockMarker.propTypes = MapView.Marker.propTypes;
	MockMapView.propTypes = MapView.propTypes;
	MockMapView.Marker = MockMarker;
	MockMapView.Callout = MockCallout;
	MockMapView.Animated = MockMapViewAnimated;
	return MockMapView;
});

test('MarkerForm renders correctly', () => {
	const tree = renderer.create(
			<MarkerForm
					initialRegion={{latitude: 1, longitude: 1, latitudeDelta: 1, longitudeDelta: 1}}
					onRegionChange={() => {}}
					
					initialTitle=""
					onTitleChange={() => {}}
					
					initialText=""
					onTextChange={() => {}}
					
					images={[]}
					onNewImage={() => {}}
					
					filters={[]}
					onFilterChange={() => {}}
			/>
	).toJSON();
	expect(tree).toMatchSnapshot();
});