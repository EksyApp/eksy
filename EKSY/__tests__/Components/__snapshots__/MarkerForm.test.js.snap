// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkerForm renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "height": 300,
        "width": "100%",
      }
    }
  >
    <MapView
      initialRegion={
        Object {
          "latitude": 1,
          "latitudeDelta": 1,
          "longitude": 1,
          "longitudeDelta": 1,
        }
      }
      onRegionChange={[Function]}
      onRegionChangeComplete={[Function]}
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <Marker
        coordinate={
          Object {
            "latitude": 1,
            "longitude": 1,
          }
        }
      />
    </MapView>
  </View>
  <View>
    <View>
      <View
        style={
          Array [
            Object {},
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "#86939e",
                  "fontSize": 15,
                  "fontWeight": "bold",
                  "marginBottom": 1,
                  "marginLeft": 20,
                  "marginRight": 20,
                  "marginTop": 15,
                },
                Object {
                  "color": "rgb(101,114,122)",
                  "fontSize": 20,
                },
                undefined,
              ],
            ]
          }
        >
          Title
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "#bdc6cf",
              "borderBottomWidth": 1,
              "marginLeft": 20,
              "marginRight": 20,
            },
            Object {
              "backgroundColor": "rgb(226,227,217)",
              "margin": 10,
            },
          ]
        }
      >
        <TextInput
          defaultValue=""
          label="Title"
          onChangeText={[Function]}
          placeholder="Title"
          selectionColor="#86939e"
          style={
            Array [
              Object {
                "color": "#86939e",
                "fontSize": 17.5,
                "height": 36,
                "width": 750,
              },
              Object {
                "color": "rgb(89,85,78)",
                "fontSize": 18,
              },
            ]
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
    <View>
      <View
        style={
          Array [
            Object {},
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "#86939e",
                  "fontSize": 15,
                  "fontWeight": "bold",
                  "marginBottom": 1,
                  "marginLeft": 20,
                  "marginRight": 20,
                  "marginTop": 15,
                },
                Object {
                  "color": "rgb(101,114,122)",
                  "fontSize": 20,
                },
                undefined,
              ],
            ]
          }
        >
          Text
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "#bdc6cf",
              "borderBottomWidth": 1,
              "marginLeft": 20,
              "marginRight": 20,
            },
            Array [
              Object {
                "backgroundColor": "rgb(226,227,217)",
                "margin": 10,
              },
              undefined,
            ],
          ]
        }
      >
        <TextInput
          defaultValue=""
          label="Text"
          multiline={true}
          onChange={[Function]}
          onChangeText={[Function]}
          placeholder="Text"
          selectionColor="#86939e"
          style={
            Array [
              Object {
                "color": "#86939e",
                "fontSize": 17.5,
                "height": 36,
                "width": 750,
              },
              Array [
                Object {
                  "color": "rgb(89,85,78)",
                  "fontSize": 18,
                },
                undefined,
                Object {
                  "height": 45,
                },
              ],
            ]
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgb(89,85,78)",
            "fontSize": 18,
            "textAlign": "center",
            "width": "100%",
          },
          undefined,
        ]
      }
    >
      
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "rgb(230,232,227)",
          "borderColor": "rgb(101,114,122)",
          "borderRadius": 5,
          "borderWidth": 1,
          "height": 40,
          "marginBottom": 15,
          "marginLeft": 5,
          "marginRight": 5,
          "marginTop": 5,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "rgb(101,114,122)",
            "fontSize": 16,
            "fontWeight": "600",
            "paddingBottom": 10,
            "paddingTop": 10,
          }
        }
      >
        Add image
      </Text>
    </View>
    <RCTScrollView
      dataSource={
        ListViewDataSource {
          "items": 0,
        }
      }
      renderRow={[Function]}
      renderScrollComponent={[Function]}
      style={undefined}
    >
      <View />
    </RCTScrollView>
    <RCTScrollView
      dataSource={
        ListViewDataSource {
          "items": 0,
        }
      }
      renderRow={[Function]}
      renderScrollComponent={[Function]}
      style={undefined}
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;
